lib/streamlit/util.py

```diff
@@ -154,3 +155,33 @@ def calc_md5(s: str) -> str:
     h = hashlib.new("md5")
     h.update(s.encode("utf-8"))
     return h.hexdigest()
+
+
+def drop_key_query_params(
+    query_params: Dict[str, List[str]], keys_to_drop: List[str]
+) -> bool:
+    """Drop query_params (passed by reference) containing keys_to_drop: List[str] (case-insensitive) query params
+    Returns True if at least one param was dropped. False when there were no keys_to_drop found.
+    """
+    found = False
+    for param_name in copy.deepcopy(query_params).keys():
+        if param_name.lower() in keys_to_drop:
+            found = True
+            query_params.pop(param_name)
+    return found
+
+
+def extract_key_query_params(query_params: Dict[str, List[str]], param_key: str) -> str:
+    """Extracts key (case-insensitive) query params from Dict,
+    and returns them as query params string."""
+    embed_values: List[str] = []
+    for param_name in query_params.keys():
+        if param_name.lower() == param_key:
+            values = query_params.get(param_name, [])
+            if values and len(values) > 0:
+                embed_values = embed_values + values
```

Comment:
```
extracted_params = set(sum([], [value.lower() for key, value in query_params.items() if key.lower() == param_key and value]))
```

Reply:
This required slightly different list comprehension, however I got your point and updated the func to use it 👍